tgui/packages/tgui/interfaces/PreferencesMenu/index.tsx

```diff
@@ -1,31 +1,82 @@
+import { Suspense, useEffect, useState } from 'react';
 import { exhaustiveCheck } from 'tgui-core/exhaustive';
+import { fetchRetry } from 'tgui-core/http';
 
+import { resolveAsset } from '../../assets';
 import { useBackend } from '../../backend';
-import { CharacterPreferenceWindow } from './CharacterPreferenceWindow';
+import { Window } from '../../layouts';
+import { logger } from '../../logging';
+import { LoadingScreen } from '../common/LoadingScreen';
+import { CharacterPreferenceWindow } from './CharacterPreferences';
+import { GamePreferenceWindow } from './GamePreferences';
 import {
   GamePreferencesSelectedPage,
   PreferencesMenuData,
-  Window,
-} from './data';
-import { GamePreferenceWindow } from './GamePreferenceWindow';
+  PrefsWindow,
+  ServerData,
+} from './types';
+import { RandomToggleState } from './useRandomToggleState';
+import { ServerPrefs } from './useServerPrefs';
 
-export const PreferencesMenu = (props) => {
+export function PreferencesMenu(props) {
+  return (
+    <Window width={920} height={770}>
+      <Window.Content>
+        <Suspense fallback={<LoadingScreen />}>
+          <PrefsWindowInner />
+        </Suspense>
```

Comment:
byond seems to just eat it while loading the massive UI so we can at least try to give the users something in the meantime